modules/rostests/apitests/win32nt/ntuser/NtUserSetTimer.c

```diff
@@ -1,18 +1,247 @@
+    if (locIndex2 == 0)
+        countErrors++;
+    if (KillTimer(NULL, locIndex2) == 0)
+        countErrors++;
+    if (locIndex1 == locIndex2)
+        countErrors++;
+
+    return (countErrors == 0);
+}
+
+// TEST WITH MESSAGES WITH WINDOW - test count of sent messages
+static BOOL testW1(HWND hwnd)
+{
+    UINT i, countErrors = 0;
+
+    if (hwnd == NULL)
+        return FALSE;
+
+    int minMessages = ((float)SLEEP_TIME / (float)TESTW1_INTERVAL) * (1 - TIME_TOLERANCE);
+    int maxMessages = ((float)SLEEP_TIME / (float)TESTW1_INTERVAL) * (1 + TIME_TOLERANCE);
+
+    ZeroMemory(timerIdW1, sizeof(timerIdW1));
+
+    for (i = 0; i < TESTW1_COUNT; i++)
+    {
+        UINT_PTR locIndex = SetTimer(hwnd, i, TESTW1_INTERVAL, NULL);
+        if (locIndex == 0)
+            countErrors++;
+    }
+
+    ULONGLONG startTime = GetTickCount();
+
+    MSG msg = { NULL };
+    while (GetMessage(&msg, NULL, 0, 0))
+    {
+        TranslateMessage(&msg);
+        DispatchMessage(&msg);
+
+        if (GetTickCount() - startTime >= SLEEP_TIME)
+            PostQuitMessage(0);
+    }
+
+    for (i = 0; i < TESTW1_COUNT; i++)
+    {
+        if ((timerIdW1[i].counter < minMessages) || (timerIdW1[i].counter > maxMessages))
+            countErrors++;
+    }
+
+    for (i = 0; i < TESTW1_COUNT; i++)
+    {
+        if (KillTimer(hwnd, i) == 0)
+            countErrors++;
+    }
+
+    return (countErrors == 0);
+}
+
+// TEST WITH MESSAGES WITH WINDOW - create many timers
+static BOOL testW2(HWND hwnd)
+{
+    UINT i, countErrors = 0;
+
+    if (hwnd == NULL)
+        return FALSE;
+
+    for (i = 0; i < TESTW2_COUNT; i++)
+    {
+        UINT_PTR result = SetTimer(hwnd, 1, TESTW2_INTERVAL, NULL);
+        if (result == 0)
+            countErrors++;
+        if (KillTimer(hwnd, 1) == 0)
+            countErrors++;
+    }
+
+    return (countErrors == 0);
+}
+
 START_TEST(NtUserSetTimer)
 {
+    // TEST WITH MESSAGES WITHOUT WINDOW - test count of sent messages
+    TEST(test1());
+
+    // TEST WITH MESSAGES WITHOUT WINDOW - create many timers
+    TEST(test2());
+
+    // TEST WITH MESSAGES WITHOUT WINDOW - test different ids
+    TEST(test3());
+
+    WNDCLASS wc = { 0 };
+    wc.lpfnWndProc = WindowProc;
+    wc.hInstance = GetModuleHandle(NULL);
+    wc.lpszClassName = "TimerWindowClass";
+    RegisterClass(&wc);
+
+    HWND hwnd = CreateWindowEx(0, "TimerWindowClass", "Timer Window", 0,
+        CW_USEDEFAULT, CW_USEDEFAULT, CW_USEDEFAULT, CW_USEDEFAULT,
+        HWND_MESSAGE, NULL, GetModuleHandle(NULL), NULL);
 
+    // TEST WITH MESSAGES WITH WINDOW - test count of sent messages
+    TEST(testW1(hwnd));
 
-    // check valid argument
-    // check for replacement / new timers
-    // check when expiries are handled (msgs and calls)
+    // TEST WITH MESSAGES WITH WINDOW - create many timers
+    TEST(testW2(hwnd));
 
+    if (hwnd != NULL)
+        DestroyWindow(hwnd);
+    UnregisterClass("TimerWindowClass", GetModuleHandle(NULL));
```

Comment:
hInstance comment:
```suggestion
    UnregisterClassW(L"TimerWindowClass", NULL);
```
2nd parameter is optional and when giving NULL it does the same thing as GetModuleHandle(NULL)

Alternatively if you want to keep GetModuleHandle explicit, I would suggest you add a
```c
HINSTANCE hInstance = GetModuleHandle(NULL);
```
at the beginning of the START_TEST(...) function, and then use that `hInstance` variable in all these places where you need it.